packages/firestore/test/integration/api/get_options.test.ts

```diff
@@ -0,0 +1,612 @@
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import { apiDescribe, withTestDoc, withTestCollection } from '../util/helpers';
+
+apiDescribe('GetOptions', persistence => {
+  it('get document while online with default get options', () => {
+    const initialData = { key: 'value' };
+    return withTestDoc(persistence, docRef => {
+      return docRef
+        .set(initialData)
+        .then(() => docRef.get())
+        .then(doc => {
+          expect(doc.exists).to.be.true;
+          expect(doc.metadata.fromCache).to.be.false;
+          expect(doc.metadata.hasPendingWrites).to.be.false;
+          expect(doc.data()).to.deep.equal(initialData);
+        });
+    });
+  });
+
+  it('get collection while online with default get options', () => {
+    const initialDocs = {
+      doc1: { key1: 'value1' },
+      doc2: { key2: 'value2' },
+      doc3: { key3: 'value3' }
+    };
+    return withTestCollection(persistence, initialDocs, colRef => {
+      return colRef.get().then(qrySnap => {
+        expect(qrySnap.metadata.fromCache).to.be.false;
+        expect(qrySnap.metadata.hasPendingWrites).to.be.false;
+        expect(qrySnap.docChanges.length).to.equal(3);
+        let docsData = {};
+        qrySnap.forEach(doc => {
+          docsData[doc.id] = doc.data();
+        });
+        expect(docsData).to.deep.equal(initialDocs);
+      });
+    });
+  });
+
+  it('get document while offline with default get options', () => {
+    const initialData = { key: 'value' };
+    const newData = { key2: 'value2' };
+    return withTestDoc(persistence, docRef => {
+      return docRef
+        .set(initialData)
+        .then(() => docRef.firestore.disableNetwork())
+        .then(() => {
+          // NB: since we're offline, the returned promise won't complete
+          docRef.set(newData);
+          return docRef.get();
+        })
+        .then(doc => {
+          expect(doc.exists).to.be.true;
+          expect(doc.metadata.fromCache).to.be.true;
+          expect(doc.metadata.hasPendingWrites).to.be.true;
+          expect(doc.data()).to.deep.equal(newData);
+        });
+    });
+  });
+
+  it('get collection while offline with default get options', () => {
+    const initialDocs = {
+      doc1: { key1: 'value1' },
+      doc2: { key2: 'value2' },
+      doc3: { key3: 'value3' }
+    };
+    return withTestCollection(persistence, initialDocs, colRef => {
+      // force local cache of these
+      return (
+        colRef
+          .get()
+          // now go offine. Note that if persistence is disabled, this will cause
+          // the initialDocs to be garbage collected.
```

Comment:
I see that this (and the above test) combat the GC'ing by doing local .set()s instead...  But this means that:
1. The initialDocs don't really matter (the test would still pass without them I believe).
2. This is really testing latency compensation instead of the cache.

Instead I think we should either:
1. Disable the test when persistence is not enabled (`if (!persistence) { return; }` or whatever)
2. Register an onSnapshot listener that includes the data. This will hold them in the cache for the duration of the listener.